### ISSUE TITLE
swup.on('event') vs document.addEventListener('swup:event')

### ISSUE THREAD

**Issue by kybradeck** [gmrchk/swup#89]
I was currently experiencing issues with Swup getting stuck at `willReplaceContent`.

I had several `swup.on('willReplaceContent', function () ...` calls in my code to handle loading and killing scripts on certain pages. Replacing them all with `document.addEventListener('swup:willReplaceContent', function () ...` immediately fixed all the problems. 

This doesn't make sense to me since I assume they would work identically, but I wanted to report it as a potential bug.

**Comment by gmrchk** [comment]
Hi, the `on`/`off` methods are relatively new but this behavior seems strange... I will look into this. Thanks for reporting!

**Comment by kybradeck** [comment]
Hey @gmrchk, after a bit of debugging, the problem was actually with one of my external libraries, not Swup, although the problem caused Swup to get stuck on transition.

I've been using this pattern to run scripts for particular pages:

```js
if (document.querySelector('element') !== null) {
  // thing.start()
  swup.on('willReplaceContent', function () {
    // thing.stop()
  })
}
```

Although using `document.addEventListener` fixed the page transitions, it was only because the `eventListener` persisted across pages. The new page would load as intended, but the console revealed many errors (elements not found, etc) because scripts continued to run from the old page.

tldr; I think `on`/`off` are working correctly, although if a function called within `on` results in an error or does not `return` as expected, Swup will get stuck, and the console will not display the error from the 3rd party.
